presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestJoinEnumerator.java

```diff
@@ -92,29 +97,73 @@ public void testDoesNotCreateJoinWhenPartitionedOnCrossJoin()
                 new LinkedHashSet<>(ImmutableList.of(p.values(a1), p.values(b1))),
                 TRUE_LITERAL,
                 ImmutableList.of(a1, b1));
-        SymbolAllocator symbolAllocator = new SymbolAllocator();
-        CachingStatsProvider statsProvider = new CachingStatsProvider(
-                queryRunner.getStatsCalculator(),
-                Optional.empty(),
-                noLookup(),
-                queryRunner.getDefaultSession(),
-                symbolAllocator.getTypes());
-        CachingCostProvider costProvider = new CachingCostProvider(
-                queryRunner.getCostCalculator(),
-                statsProvider,
-                Optional.empty(),
-                noLookup(),
-                queryRunner.getDefaultSession(),
-                symbolAllocator.getTypes());
         ReorderJoins.JoinEnumerator joinEnumerator = new ReorderJoins.JoinEnumerator(
-                queryRunner.getDefaultSession(),
-                costProvider,
                 new CostComparator(1, 1, 1),
-                idAllocator,
                 multiJoinNode.getFilter(),
-                noLookup());
+                makeContext());
         JoinEnumerationResult actual = joinEnumerator.createJoinAccordingToPartitioning(multiJoinNode.getSources(), multiJoinNode.getOutputSymbols(), ImmutableSet.of(0));
         assertFalse(actual.getPlanNode().isPresent());
         assertEquals(actual.getCost(), INFINITE_COST);
     }
+
+    private Rule.Context makeContext()
+    {
+        return new Rule.Context()
+        {
+            private PlanNodeIdAllocator planNodeIdAllocator = new PlanNodeIdAllocator();
```

Comment:
Nit: Move the fields declaration outside the inner class. (https://github.com/prestodb/presto/pull/11009/commits/2fcdd5f840997c66dd9e76e1a9eaf87bc7f9b6b7#diff-5eed8b9c2cc6e5f6be8077b8c2328943R194)